src/scripts/components/Chart.jsx

```diff
@@ -1,66 +1,104 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import ReactHtmlParser from 'react-html-parser';
 import { Bar, Doughnut, Line } from 'react-chartjs-2';
 
 import { Icon } from '.';
 import Tooltip from './Tooltip';
 
-const renderChart = (opts) => {
-  switch (opts.type.toLowerCase()) {
-    case 'bar':
-      return (<Bar {...opts} />);
-    case 'doughnut':
-      return (<Doughnut {...opts} />);
-    case 'line':
-      return (<Line {...opts} />);
-    default:
-      return (<Bar {...opts} />);
-  }
-};
+class Chart extends React.Component {
+  state = {
+    isChartData: false,
+  };
 
-const renderNoData = () => (
-  <div className="chart__without-data">
-      Sorry! There&apos;s nothing to show.
-    <p>
-      Once data is available for this section, it will appear here.
-    </p>
-  </div>
-);
+  legendCallback = function(chart) {  // eslint-disable-line
+    const { labels } = chart.data;
+    const { data } = chart.data.datasets[0];
+    const { backgroundColor } = chart.data.datasets[0];
+    const text = [];
+    text.push('<ul>');
+    for (let i = 0; i < labels.length; i++) {
+      text.push('<li>');
+      text.push(`<div style="border-color: ${backgroundColor[i]}"><span class="list-title" /> <span>${labels[i]}</span><span>${data[i]}</span></div>`);
+      text.push('</li>');
+    }
+    text.push('</ul>');
+    return text.join('');
+  };
 
-const Chart = (props) => {
-  const { ...opts } = props;
-  let isChartData = false;
-  if (typeof opts.data === 'undefined' || opts.data === null || Object.keys(opts.data).length === 0 || Object.keys(opts.data.datasets[0].data).length === 0) {
-    isChartData = true;
+  componentWillMount() {
+    const { ...opts } = this.props;
```

Comment:
Looks like you're only using `this.props.data`, so only destructure that property.

Reply:
done